src/main/java/com/flippingutilities/model/RecipeFlip.java

```diff
@@ -0,0 +1,128 @@
+import com.flippingutilities.utilities.RecipeItem;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+
+import java.time.Instant;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * This is the structure that represents a flip made out of various different items, either because
+ * those items were combined to make some other one, or were the result of breaking some other item.
+ * Ex: guthan set bought then broken up into pieces and sold.
+ * The "parent" in these recipeFlips is not necessarily the output item (of combining or breaking stuff), it is
+ * the item that I have arbitrarily chosen to own the profits of the recipe flips. This is because profits cannot
+ * be divided over all the items in the recipe flip. What decides how much profit each item gets? For that reason
+ * I've decided that each recipe has a parent item (the parent item that will end up in the recipeFlip) and that
+ * parent item shall own the profits, revenue, expense, etc for the recipe flip.
+ */
+@Data
+@AllArgsConstructor
+public class RecipeFlip {
+    Instant timeOfCreation;
+    Map<Integer, Map<String, PartialOffer>> outputs;
+    //item id to a map of offer id to offer
+    Map<Integer, Map<String, PartialOffer>> inputs;
+
+    public RecipeFlip(Recipe recipe, Map<Integer, Map<String, PartialOffer>> allPartialOffers) {
+        Set<Integer> recipeInputIds = recipe.getInputIds();
+        Set<Integer> recipeOutputIds = recipe.getOutputIds();
+
+        this.timeOfCreation = Instant.now();
+        this.inputs = allPartialOffers.entrySet().stream().filter(e -> recipeInputIds.contains(e.getKey()))
+            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+        this.outputs = allPartialOffers.entrySet().stream().filter(e -> recipeOutputIds.contains(e.getKey()))
+            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+    }
+
+    public RecipeFlip clone() {
+        Instant clonedInstant = Instant.ofEpochSecond(timeOfCreation.getEpochSecond());
+        Map<Integer, Map<String, PartialOffer>> clonedOutputs = cloneComponents(outputs);
+        Map<Integer, Map<String, PartialOffer>> clonedInputs = cloneComponents(inputs);
+        return new RecipeFlip(clonedInstant, clonedOutputs, clonedInputs);
+    }
+
+    private Map<Integer, Map<String, PartialOffer>> cloneComponents(Map<Integer, Map<String, PartialOffer>> component) {
+        return component.entrySet().stream()
+            .map(e -> Map.entry(
+                e.getKey(),
+                e.getValue().entrySet().stream()
+                    .map(entry -> Map.entry(entry.getKey(), entry.getValue().clone()))
+                    .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue))))
+            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+    }
+
+    public long getProfit() {
+        return getRevenue() - getExpense();
+    }
+
+    private long getExpenseOrRevenue(boolean getExpense) {
```

Comment:
method name: `getIngredientsValue(isBuyOffer)`?
I would even make an enum with true/false for the clear name, but renaming `getExpense` to `isBuyOffer` should be better already imo